### PR TITLE
DurableTask.Redis: Initial orchestration support

### PR DESCRIPTION
**Pull request by ConnorMcMahon**
Rough draft support for basic orchestrations. Have tested simple orchestration and fairly large Fan-in/Fan-out.

### CODE REVIEW

File: Test/DurableTask.AzureStorage.Tests/DurableTask.AzureStorage.Tests.csproj [hunk 1]
```diff
@@ -2,7 +2,7 @@
 
   <Import Project="$([MSBuild]::GetDirectoryNameOfFileAbove($(MSBuildThisFileDirectory),DurableTask.sln))\tools\DurableTask.props" />
   <PropertyGroup>
-    <TargetFramework>net451</TargetFramework>
+    <TargetFrameworks>net451</TargetFrameworks>
```
Comment (ConnorMcMahon) [inline Test/DurableTask.AzureStorage.Tests/DurableTask.AzureStorage.Tests.csproj]: This is leftover from an attempt to reuse test code from this class. I will revert it.

File: Test/DurableTask.Redis.Tests/TestOrchestrationClient.cs [hunk 2]
```diff
@@ -0,0 +1,143 @@
+﻿//  ----------------------------------------------------------------------------------
```
Comment (ConnorMcMahon) [inline Test/DurableTask.Redis.Tests/TestOrchestrationClient.cs]: This entire class was copied wholesale from DurableTask.AzureStorage.Tests. Trying to get that project as a dependency for this project was taking too much time, so I just copied it.

**Comment by cgillum** [comment]
Adding @gled4er for FYI

File: Test/DurableTask.Redis.Tests/TestOrchestrationClient.cs (hunk 2 above)
Comment (cgillum) [reply Test/DurableTask.Redis.Tests/TestOrchestrationClient.cs]: Might be worth adding a comment explaining this.

Comment (ElanHasson) [reply Test/DurableTask.Redis.Tests/TestOrchestrationClient.cs]: Possibly adding as a linked file?

**Pull request merged**